SConstruct

```diff
@@ -141,6 +142,15 @@ def RecursiveGlob(pattern, dir_name=buildDirectory):
 	matches = [i for i in matches if not '{}main.cpp'.format(os.path.sep) in str(i)]
 	return matches
 
+try:
+    env.Tool('compilation_db')
+    env.CompilationDatabase()
+# scons before 4.0.0 is used. In that case, simply don't provide a compilation database.
+except EnvironmentError:
+    pass
+except SConsEnvironmentError:
+    pass
```

Comment:
```suggestion
except (EnvironmentError, SConsEnvironmentError):
    pass
```

Or use the `suppress` context handler since the only goal is to ignore those exceptions if they happen. (That requires python 3.4 and I don't remember the exact version used by steam, though.)

Reply:
I can't write it like that since the `SConsEnvironmentError` type doesn't exist in the AppImage CD. I'm basically abusing the fact that Python is a dynamic language.

Reply:
you can now :)